### PR TITLE
fix: logic updated for updating application status

### LINKED ISSUE #5211: Bug: application status changes to healthy on unhibernating, even if unhibernation fails
### 📜 Description

when you unhibernate an already hibernating aplication, and for some reason the deployment(unhibernation) fails, in that case application status gets updated to `healthy` with 0 pods.

### 👟 Reproduction steps

unhibernate an already hibernated application, by tweaking API_TIMEOUT

### 👍 Expected behavior

application status should still be `hibernating`

### 👎 Actual Behavior

status changes to `healthy`, even after failed deployment

### ☸ Kubernetes version

.

### Cloud provider

<details>
.
</details>

### 🌍 Browser

Chrome

### 🧱 Your Environment

_No response_

### ✅ Proposed Solution

_No response_

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find any similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/devtron-labs/devtron/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by ashishdevtron**
# Description

logic updated to showing and updating application status, currently we were only checking last deployment type but we also have to check whether the deployment was successful or not

Fixes #5211

## Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code.
* [ ] I have commented my code, particularly in hard-to-understand areas.
* [ ] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: internal/sql/repository/chartConfig/PipelineOverrideRepository.go [hunk 1]
```diff
@@ -188,6 +194,30 @@ func (impl PipelineOverrideRepositoryImpl) GetLatestRelease(appId, environmentId
...
+	var override PipelineOverride
+	var latestDeployment LatestDeployment
+	err := impl.dbConnection.Model(&override).
+		Column("pipeline_override.deployment_type", "cwr.status").
+		Join("join pipeline p on pipeline_override.pipeline_id = p.id").
+		Join("join cd_workflow cw on pipeline_override.cd_workflow_id = cw.id").
+		Join("join cd_workflow_runner cwr on cwr.cd_workflow_id = cw.id").
+		Where("p.app_id = ?", appId).
+		Where("p.environment_id = ?", environmentId).
+		Where("cwr.status in (?)", pg.In([]string{pipelineConfig.WorkflowSucceeded, pipelineConfig.WorkflowInProgress})).
```
Comment (kartik-579) [inline internal/sql/repository/chartConfig/PipelineOverrideRepository.go:209]: unableToFetch, timedOut are also non failed, to change condition or method name

File: api/restHandler/app/appList/AppListingRestHandler.go [hunk 2]
```diff
@@ -990,13 +992,28 @@ func (handler AppListingRestHandlerImpl) getAppDetails(ctx context.Context, appI
 	return appDetail, err, appId
 }
 
+func getApplicationStatusFromLatestDeployment(latestDeployment *chartConfig.LatestDeployment) string {
+	if latestDeployment != nil {
+		if latestDeployment.RunnerStatus == argoApplication.Progressing {
```
Comment (kartik-579) [inline api/restHandler/app/appList/AppListingRestHandler.go:997]: if the last non failed(definition according to db method name) is in progressing we cannot ensure it went ahead for deployment, need to check the actual latest runner to dedupe and timeline

File: pkg/bulkAction/BulkUpdateService.go [hunk 3]
```diff
@@ -964,6 +967,10 @@ func (impl BulkUpdateServiceImpl) BulkUpdate(bulkUpdatePayload *BulkUpdatePayloa
 	return bulkUpdateResponse
 }
 
+func needToSkipDeployment(status string) bool {
+	return status != pipelineConfig.WorkflowFailed && status != pipelineConfig.WorkflowUnableToFetchState && status != pipelineConfig.WorkflowTimedOut
```
Comment (kartik-579) [inline pkg/bulkAction/BulkUpdateService.go:971]: need product call on this

**Comment by vikramdevtron** [comment]
Not required,  implementation is incorrect

**Pull request closed without merging**